env0/data_sshkey.go

```diff
@@ -31,23 +32,23 @@ func dataSshKey() *schema.Resource {
 
 func dataSshKeyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	name, nameSpecified := d.GetOk("name")
-	var sshKey client.SshKey
-	var err diag.Diagnostics
+	var sshKey *client.SshKey
+	var err error
 	if nameSpecified {
 		sshKey, err = getSshKeyByName(name, meta)
-		if err != nil {
-			return err
-		}
 	} else {
 		id, idSpecified := d.GetOk("id")
 		if !idSpecified {
 			return diag.Errorf("At lease one of 'id', 'name' must be specified")
 		}
 		sshKey, err = getSshKeyById(id, meta)
-		if err != nil {
-			return err
+		if err == nil && sshKey == nil {
+			err = fmt.Errorf("id %s not found", id)
```

Comment:
Can you please add some context to the error text? (`ssh key with id X not found`)